### PR TITLE
Allow disabling purchase AVS/CVN policy on purchase

### PR DESCRIPTION
**Pull request by brendanpeters**
Adds ability to set whether to ignore AVS/CVN policies in `PurchaseRequest`.

### CODE REVIEW

File: src/Message/PurchaseRequest.php [hunk 1]
```diff
@@ -62,13 +62,41 @@ protected function getFunction()
...
     }
 
-    public function getData($paymentMethodType = self::PAYMENT_METHOD_CREDIT_CARD)
+    public function getData(string $paymentMethodType = self::PAYMENT_METHOD_CREDIT_CARD)
     {
         $data = parent::getData($paymentMethodType);
 
-        $data['ignoreAvsPolicy'] = false;
-        $data['ignoreCvnPolicy'] = false;
+        $ignore_avs = $this->getIgnoreAvsPolicy();
```
Comment (DanayaMel) [inline src/Message/PurchaseRequest.php]: Could you update the doc block above to include this param, now that it isn't set by default? And a short comment about what this is for?

File: src/Message/PurchaseRequest.php [hunk 2]
```diff
@@ -62,13 +62,41 @@ protected function getFunction()
         return 'authCapture';
     }
 
-    public function getData($paymentMethodType = self::PAYMENT_METHOD_CREDIT_CARD)
+    public function getData(string $paymentMethodType = self::PAYMENT_METHOD_CREDIT_CARD)
```
Comment (jcmanzo) [inline src/Message/PurchaseRequest.php]: This repo supports PHP 5.3+ and scalar type hinting is a PH 7+ feature so this repo needs to rely on PHPdocs

Comment (brendanpeters) [reply src/Message/PurchaseRequest.php]: bummer. :( thanks, I'll change that

**Comment by brendanpeters** [comment]
@DanayaMel updated the docblock
@jcmanzo removed scalar type hints

File: src/Message/PurchaseRequest.php [hunk 3]
```diff
@@ -62,13 +66,51 @@ protected function getFunction()
...
+    /**
+     * @return null|bool
+     */
+    public function getIgnoreAvsPolicy()
+    {
+        return $this->getParameter('ignoreAvsPolicy');
+    }
+
+    /**
+     * @param bool $ignore
```
Comment (jcmanzo) [inline src/Message/PurchaseRequest.php]: Psalm will complain about not having an `@return` here actually

File: src/Message/PurchaseRequest.php [hunk 4]
```diff
@@ -62,13 +66,51 @@ protected function getFunction()
...
+     * @return null|bool
+     */
+    public function getIgnoreCvnPolicy()
+    {
+        return $this->getParameter('ignoreCvnPolicy');
+    }
+
+    /**
+     * @param bool $ignore
+     */
```
Comment (jcmanzo) [inline src/Message/PurchaseRequest.php]: Psalm will complain about not having an @return here

File: src/Message/PurchaseRequest.php (hunk 3 above)
Comment (DanayaMel) [reply src/Message/PurchaseRequest.php]: I think you have to do:
```
@return static
```

**Pull request merged**